config/routes.rb

```diff
@@ -761,17 +762,24 @@
 
     scope "/topics", username: RouteFormat.username do
       get "created-by/:username" => "list#topics_by", as: "topics_by", defaults: { format: :json }
+      get "private-messages-all/:username" => "list#private_messages_all", as: "topics_private_messages_all", defaults: { format: :json }
+      get "private-messages-all-sent/:username" => "list#private_messages_all_sent", as: "topics_private_messages_all_sent", defaults: { format: :json }
+      get "private-messages-all-new/:username" => "list#private_messages_all_new", as: "topics_private_messages_all_new", defaults: { format: :json }
+      get "private-messages-all-unread/:username" => "list#private_messages_all_unread", as: "topics_private_messages_all_unread", defaults: { format: :json }
+      get "private-messages-all-archive/:username" => "list#private_messages_all_archive", as: "topics_private_messages_all_archive", defaults: { format: :json }
       get "private-messages/:username" => "list#private_messages", as: "topics_private_messages", defaults: { format: :json }
       get "private-messages-sent/:username" => "list#private_messages_sent", as: "topics_private_messages_sent", defaults: { format: :json }
       get "private-messages-archive/:username" => "list#private_messages_archive", as: "topics_private_messages_archive", defaults: { format: :json }
       get "private-messages-unread/:username" => "list#private_messages_unread", as: "topics_private_messages_unread", defaults: { format: :json }
-      get "private-messages-tags/:username/:tag_id.json" => "list#private_messages_tag", as: "topics_private_messages_tag", defaults: { format: :json }
```

Comment:
Remove this method without depreciation because we've never listed this as a public API: https://docs.discourse.org/

Reply:
We should be careful taking this approach, because that doesn't guarantee anyone is not relying on it anyway. For better or worse we have recommended in the past that people look at XHR requests and use those.

Reply:
I've kept the route for now and added a comment to deprecate it after Discourse 2.9 has been released.